### DISCUSSION TITLE
Problem installing npm package

### DISCUSSION THREAD

**Discussion by JoepKockelkorn** [johnlindquist/kit#323] [Errors] [answered]
I'm getting the following error after being prompted to install an npm package and confirming to install:

```
[2021-06-19 21:00:09.081] [info]  *** /Users/joepkockelkorn/.kit/cli/error-action.js open-hatseflats-channel.js,Error: /Users/joepkockelkorn/.kenv/node_modules/@nut-tree/nut-js/package.json doesn't exist
Error: Error: /Users/joepkockelkorn/.kenv/node_modules/@nut-tree/nut-js/package.json doesn't exist
    at kenvImport (file:///Users/joepkockelkorn/.kit/api/npm.js:43:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async file:///Users/joepkockelkorn/.kit/api/npm.js:58:12
    at async file:///Users/joepkockelkorn/.kenv/scripts/open-hatseflats-channel.js?uuid=62e4eb0e-1462-4168-9a0a-a1d33145e053:6:23,/Users/joepkockelkorn/.kenv/scripts/open-hatseflats-channel.js,1,1 ***
```

It seems the package is not installed properly? The script content is this:

```javascript
const { clipboard } = await npm("@nut-tree/nut-js");
await clipboard.paste();
```

Is there a fix for it, or a workaround I didn't spot yet?

**Comment by johnlindquist** [comment, marked as answer]
Hi @JoepKockelkorn ,

I found this issue on the nut-js repo: https://github.com/nut-tree/nut.js/issues/223

The "solution" was to downgrade to node 15, which isn't really an option (Kit is on node 16/npm 7).

I'm out of town this weekend, so can't really dig into this until next week at my home computer. I guess you could try installing node 15 then manually install in the terminal like so:

```bash
cd ~/.kenv
/wherever/node/15/is/bin/npm i @nut-tree/nut-js
```

Then use `import` instead of `npm`:

```js
const { clipboard } = await import("@nut-tree/nut-js");
await clipboard.paste();
```

That _might_ work. I'll dig in deeper next week.

**Reply by JoepKockelkorn** [reply]
Thanks for the swift reply. I'm perfectly fine with pasting myself in the meanwhile, so don't put this at the top of your list 👍

**Reply by johnlindquist** [reply]
If you're only looking to paste from the clipboard, use the global:

```js
await setSelectedText()
```

It will trigger a paste using AppleScript to cmd+v

> Note: This is different from the global `paste()` helper which reads the clipboard into the script.

**Reply by JoepKockelkorn** [reply]
I can confirm this works: `await setSelectedText(await paste());`

Thanks!
